### PR TITLE
chore: add vscode config file

### PR DESCRIPTION
**Pull request by ruibaby**
添加 vscode 基于项目的配置文件和插件推荐配置文件。

Signed-off-by: Ryan Wang <i@ryanc.cc>

### CODE REVIEW

**Comment by guqing** [comment]
要不要加一个 "editor.rulers": [80], 列宽80

File: .gitignore [hunk 1]
```diff
@@ -15,7 +15,6 @@ pnpm-debug.log*
 
 # Editor directories and files
 .idea
-.vscode
```
Comment (JohnNiang) [inline .gitignore:18]: I'd prefer to not include `.vscode` folder. We could add `.vscode/extensions.json` and `.vscode/settings.json` files into git forcely event if we exclude `.vscode` folder.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (JohnNiang) [reply .gitignore:18]: Since there might have other unexpected files in `.vscode`, such as `.vscode/launch.json` and `.vscode/tasks.json`.

**Comment by ruibaby** [comment]
> 要不要加一个 "editor.rulers": [80], 列宽80

这个通过 prettier 控制了。

**Review by guqing — APPROVED** [review]

**Review by JohnNiang — APPROVED** [review]
LGTM

**Pull request merged**